lab_3_ann_retriever/main.py

```diff
@@ -157,6 +197,22 @@ def build(self) -> bool:
         Returns:
             bool: True if built successfully, False in other case
         """
+        if not self._corpus:
+            return False
+        unique_words = set()
+        for words in self._corpus:
+            if not isinstance(words, list) or not words:
+                return False
+            unique_words.update(words)
+        self._vocabulary = sorted(list(unique_words))
+        self._token2ind = {token: ind for ind, token in enumerate(self._vocabulary)}
+        idf = calculate_idf(self._vocabulary, self._corpus)
+        if isinstance(idf, dict):
```

Comment:
remove this check, just check this it next if statement and after this assign to `self._idf_values`

```
idf = calculate_idf
if not blablabla or not isinstance(idf, dict) or not blabla:
    return False
self._idf_values = idf
return True
```
What do you think about this example?